src/bindings/c/include/openvino/c/ov_infer_request.h

```diff
@@ -70,37 +71,149 @@ OPENVINO_C_API(ov_status_e)
+ * @brief Sets an input/output tensor to infer request for the port.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param port Const port of the input or output tensor, which can be got by call interface from ov_model_t/ov_compiled_model_t.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_tensor_by_const_port(ov_infer_request_t* infer_request, const ov_output_const_node_t* port, const ov_tensor_t* tensor);
+
+/**
+ * @brief Sets an input tensor to infer on by the index of tensor.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param idx Index of the input port. If @p idx is greater than the number of model inputs, an exception is thrown.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_input_tensor_by_index(ov_infer_request_t* infer_request, const size_t idx, const ov_tensor_t* tensor);
+
+/**
+ * @brief Sets an input tensor for the model with single input to infer on.
+ * @note If model has several inputs, an exception is thrown.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_input_tensor(ov_infer_request_t* infer_request, const ov_tensor_t* tensor);
+
+/**
+ * @brief Sets an output tensor to infer by the index of output tensor.
+ * @note Index of the output preserved accross ov_model_t, ov_compiled_model_t.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param idx Index of the output tensor.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_output_tensor_by_index(ov_infer_request_t* infer_request, const size_t idx, const ov_tensor_t* tensor);
+
+/**
+ * @brief Sets an output tensor to infer models with single output.
+ * @note If model has several outputs, an exception is thrown.
  * @ingroup infer_request
  * @param infer_request A pointer to the ov_infer_request_t.
- * @param tensor_name  Name of the input or output tensor.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_output_tensor(ov_infer_request_t* infer_request, const ov_tensor_t* tensor);
+
+/**
+ * @brief Gets an input/output tensor by the name of tensor.
+ * @note If model has several outputs, an exception is thrown.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param tensor_name Name of the input or output tensor to get.
  * @param tensor Reference to the tensor.
```

Comment:
Pointer to tensor ptr?

Reply:
ov_tensor_t is hidden structure, we should use ov_tensor_t * to access it, and this API will get a ov_tensor_t *, so we will set ov_tensor_t **.

Reply:
I mean that it is not a reference :)

Reply:
yes, it is to return one